python/ray/experimental/state.py

```diff
@@ -304,3 +304,33 @@ def client_table(self):
       node_info[node_ip_address].append(client_info_parsed)
 
     return node_info
+
+  def log_table(self):
+    """Fetch and return a dictionary of log file names to outputs, sorted by
+    IP address.
+
+    Returns:
+      IP address to log file name to log file contents mappings.
+    """
+    self._check_connected()
+    r = ray.worker.global_worker.redis_client
+    relevant_files = r.keys("LOGFILE*")
+
+    ip_filename_file = dict()
+
+    for filename in relevant_files:
+      filename = filename.decode('utf8')
+      filename_components = filename.split(":")
+      ip_addr = filename_components[1]
+      fileaddr = filename_components[2]
+
+      with open(fileaddr, encoding='utf8') as f:
```

Comment:
We probably shouldn't open the files and read them (since they may be on different machines in the cluster setting). The file contents should ideally already be in Redis (the process that puts them in Redis is this one https://github.com/ray-project/ray/blob/master/python/ray/log_monitor.py).

So you should be able to read them with something like

```python
file_contents = self.redis_client.lrange(filename, 0, -1)
```